cachi2/core/models/sbom.py

```diff
@@ -93,3 +93,133 @@ class Sbom(pydantic.BaseModel):
     def _unique_components(cls, components: list[Component]) -> list[Component]:
         """Sort and de-duplicate components."""
         return unique_sorted(components, by=lambda component: component.key())
+
+    def to_spdx(self) -> "SPDXSbom":
+        """Convert a CycloneDX SBOM to an SPDX SBOM."""
+        return SPDXSbom(
+            packages=[
+                SPDXPackage(
+                    name=component.name,
+                    version=component.version,
+                    package_download_location=component.purl,
+                )
+                for component in self.components
+            ],
+            creationInfo=SPDXCreationInfo(
+                creators=[f"{tool.vendor} {tool.name}" for tool in self.metadata.tools]
+            ),
+        )
+
+
+class SPDXPackageExternalRef(pydantic.BaseModel):
+    """SPDX Package External Reference.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: str
+    referenceLocator: str
+    referenceType: str
+
+    def __hash__(self) -> int:
+        return hash((self.referenceLocator, self.referenceType, self.referenceCategory))
+
+
+class SPDXPackage(pydantic.BaseModel):
+    """SPDX Package.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/
+    """
+
+    name: str
+    version: Optional[str] = None
+    externalRefs: list[SPDXPackageExternalRef] = []
+
+    def key(self) -> tuple[str, ...]:
+        """Uniquely identifies a package.
+
+        Used mainly for sorting and deduplication.
+        """
+        purls = []
+        for ref in self.externalRefs:
+            if ref.referenceType == "purl":
+                purls.append(ref.referenceLocator)
+        return tuple(purls)
+
+    @classmethod
+    def from_package_dict(cls, package: dict[str, Any]) -> "SPDXPackage":
+        """Create a SPDXPackage from a Cachi2 package dictionary."""
+        if package.get("externalRefs", None):
```

Comment:
PEP8 nitpick: if you mean to check if something is `None` then you better do it wih `is`. `{}.get()` returns `None` by default, you could have replaced it with `[]` and have gotten rid of `if...else` completely.